internal/mode/static/nginx/config/main_config.go

```diff
@@ -32,3 +38,88 @@ func executeMainConfig(conf dataplane.Configuration) []executeResult {
 
 	return results
 }
+
+type mgmtConf struct {
+	Endpoint          string
+	Resolver          string
+	LicenseTokenFile  string
+	DeploymentCtxFile string
+	CACertFile        string
+	ClientSSLCertFile string
+	ClientSSLKeyFile  string
+	SkipVerify        bool
+}
+
+// generateMgmtFiles generates the NGINX Plus configuration file for the mgmt block. As part of this,
+// it writes the secret and deployment context files that are referenced in the mgmt block.
+func (g GeneratorImpl) generateMgmtFiles(conf dataplane.Configuration) []file.File {
+	if !g.plus {
+		return nil
+	}
+
+	tokenFile := file.File{
+		Content: conf.AuxiliarySecrets[graph.PlusReportJWTToken],
+		Path:    secretsFolder + "/license.jwt",
+		Type:    file.TypeSecret,
+	}
+	files := []file.File{tokenFile}
+
+	cfg := mgmtConf{
+		Endpoint:         g.usageReportConfig.Endpoint,
+		Resolver:         g.usageReportConfig.Resolver,
+		LicenseTokenFile: tokenFile.Path,
+		SkipVerify:       g.usageReportConfig.SkipVerify,
+	}
+
+	if content, ok := conf.AuxiliarySecrets[graph.PlusReportCACertificate]; ok {
+		caFile := file.File{
+			Content: content,
+			Path:    secretsFolder + "/mgmt-ca.crt",
+			Type:    file.TypeSecret,
+		}
+		cfg.CACertFile = caFile.Path
+		files = append(files, caFile)
+	}
+
+	if content, ok := conf.AuxiliarySecrets[graph.PlusReportClientSSLCertificate]; ok {
+		certFile := file.File{
+			Content: content,
+			Path:    secretsFolder + "/mgmt-tls.crt",
+			Type:    file.TypeSecret,
+		}
+		cfg.ClientSSLCertFile = certFile.Path
+		files = append(files, certFile)
+	}
+
+	if content, ok := conf.AuxiliarySecrets[graph.PlusReportClientSSLKey]; ok {
+		keyFile := file.File{
+			Content: content,
+			Path:    secretsFolder + "/mgmt-tls.key",
+			Type:    file.TypeSecret,
+		}
+		cfg.ClientSSLKeyFile = keyFile.Path
+		files = append(files, keyFile)
+	}
+
+	depCtx, err := json.Marshal(conf.DeploymentContext)
+	if err != nil {
+		g.logger.Error(err, "error building deployment context for mgmt block")
```

Comment:
Should we have a default `deployment_ctx.json` with integration set to "ngf" that we can use if we error here? Does that provide any value?